src/backend/routers/iep.test.ts

```diff
@@ -88,6 +103,62 @@ test("basic flow - add/get goals, subgoals, tasks", async (t) => {
   );
 });
 
+test("addTask - no duplicate subgoal_id + assigned_id combo", async (t) => {
+  const { trpc, seed } = await getTestServer(t, {
+    authenticateAs: "case_manager",
+  });
+
+  const para_id = seed.para.user_id;
+
+  const iep = await trpc.student.addIep.mutate({
+    student_id: seed.student.student_id,
+    start_date: new Date("2023-01-01"),
+    end_date: new Date("2023-12-31"),
+  });
+
+  const goal1 = await trpc.iep.addGoal.mutate({
+    iep_id: iep.iep_id,
+    description: "goal 1",
+    category: "writing",
+  });
+
+  const subgoal1 = await trpc.iep.addSubgoal.mutate({
+    goal_id: goal1!.goal_id,
+    status: "Complete",
+    description: "subgoal 1",
+    setup: "",
+    instructions: "",
+    materials: "materials",
+    target_level: 100,
+    baseline_level: 20,
+    metric_name: "words",
+    attempts_per_trial: 10,
+    number_of_trials: 30,
+  });
+  const subgoal1Id = subgoal1!.subgoal_id;
+
+  await trpc.iep.addTask.mutate({
+    subgoal_id: subgoal1Id,
+    assignee_id: para_id,
+    due_date: new Date("2023-12-31"),
+    trial_count: 5,
+  });
+
+  const error = await t.throwsAsync(async () => {
+    await trpc.iep.addTask.mutate({
+      subgoal_id: subgoal1Id,
+      assignee_id: para_id,
+      due_date: new Date("2024-03-31"),
+      trial_count: 1,
+    });
+  });
+
+  t.is(
+    error?.message,
+    'duplicate key value violates unique constraint "uc_task"'
```

Comment:
I'd suggest adding more detail to this error message.

Reply:
Done!